### PR TITLE
20 sklearn api comp test

### PR DESCRIPTION
**Pull request by cklyne**
Added SKlearn tests to test suite + adjusted code to pass them.
Partially jitted code.

### CODE REVIEW

File: bde/utils/__init__.py [hunk 1]
```diff
@@ -7,12 +7,15 @@
...
 - `configs`: Contains configuration management utilities for the BDE framework.
+- `utils`: Contains general utility functions used by other modules.
 """
 
 from bde.utils import (
     configs,
+    utils,
 )
 
 __al__ = [
```
Comment (SteffenJw) [inline bde/utils/__init__.py:18]: does it need to be __all__?

File: bde/ml/__init__.py [hunk 2]
```diff
@@ -25,11 +26,13 @@
 from bde.ml import (
     loss,
     models,
-    training
+    training,
+    datasets,
 )
 
 __al__ = [
```
Comment (SteffenJw) [inline bde/ml/__init__.py:33]: should be __all__?

File: bde/ml/models.py [hunk 3]
```diff
@@ -188,18 +347,32 @@ def fit(
...
-        self.history_ = dict()
-        if self.metrics is None:
-            self.metrics = list()
-        else:
+        if y is None:
+            y = X
+        bde.utils.utils.check_fit_input(X, y)
+        metrics = [] if self.metrics is None else self.metrics
+        if len(metrics) > 0:
             raise ValueError(f"Metrics are not yet supported.")  # TODO: Remove after implementation
```
Comment (SteffenJw) [inline bde/ml/models.py:355]: rather not implemented yet error :)

**Review by SteffenJw — APPROVED** [review]

File: bde/utils/__init__.py (hunk 1 above)
Comment (cklyne) [reply bde/utils/__init__.py:18]:
```suggestion
__all__ = [
```

File: bde/ml/__init__.py (hunk 2 above)
Comment (cklyne) [reply bde/ml/__init__.py:33]:
```suggestion
__all__ = [
```

File: bde/ml/models.py (hunk 3 above)
Comment (cklyne) [reply bde/ml/models.py]: Implemented in 9aab1cd421656f51cb794a001351fb2e2011fb52.

**Pull request merged**